src/v/datalake/translation/deps.cc

```diff
@@ -577,4 +590,59 @@ translation_context::make_default_translation_context(
       std::move(probe));
 }
 
+// note: this is a temporary implementation to get the code compiling
+// to be filled with a proper implementation that tracks windows based
+// on batch timestamps + heuristics
+// Even the interface is temporary and subject to change depending on
+// the changes needed to track windows.
+class partition_lag_tracker : public translation_lag_tracker {
+public:
+    explicit partition_lag_tracker(
+      ss::lw_shared_ptr<cluster::partition> partition,
+      cluster::topic_table& topics)
+      : _partition(std::move(partition))
+      , _topics(topics)
+      , _current_window_end(scheduling::clock::now()) {
+        reset_to_next_window();
+    }
+
+    bool should_finish_inflight_translation() final {
+        auto window_ended = scheduling::clock::now() >= _current_window_end;
+        if (window_ended) {
+            reset_to_next_window();
+        }
+        return window_ended;
+    }
+
+private:
+    scheduling::clock::duration target_lag() const {
+        const auto& topic_cfg = _topics.get_topic_cfg(
+          model::topic_namespace_view{_partition->ntp()});
+        auto default_lag
+          = config::shard_local_cfg().iceberg_target_lag_ms.value();
+        if (!topic_cfg.has_value()) {
+            return std::chrono::duration_cast<scheduling::clock::duration>(
+              default_lag);
+        }
+        auto target_lag = topic_cfg->properties.iceberg_target_lag_ms.value_or(
+          default_lag);
+        return std::chrono::duration_cast<scheduling::clock::duration>(
+          target_lag);
+    }
+
+    void reset_to_next_window() { _current_window_end += target_lag(); }
```

Comment:
Curious why not now + target_lag()? Though i guess it doesn't matter if this is all throwaway